katsdpcal/reduction.py

```diff
@@ -845,9 +854,56 @@ def pipeline(data, ts, parameters, solution_stores, stream_name, sensors=None):
 
     return target_slices, av_corr
 
+def get_offsets(ts, target, t_stamps, temp, pres, humi, parameters):
+
+
+    """Calculate offset co-ordinates relative to target for each pointing.
+
+    Parameters
+    ----------
+    ts : :class:`katsdptelstate.TelescopeState`
+        Telescope state, scoped to the cal namespace within the capture block
+    target : :class:`katpoint.Target`
+        Pointing calibrator as a katpoint target object
+    t_stamps : list
+        Timestamps at which offset co-ordinates must be determined
+    temp, pres, humi : :class: 'float'
+        Atmospheric conditions used for refraction correction
+    parameters : dict
+        The pipeline parameters
+
+    Returns
+    -------
+    offsets : list of (x,y) co-ordinates for each pointing
+    """
+    rc = katpoint.RefractionCorrection()
+    # Set refant index
+    refant_ind = parameters['refant_index']
+    refant = parameters['antennas'][refant_ind]
+    # Get middle timestamp of each track slice
+    offsets=[]
+    for j in t_stamps:
+        # AZ/EL co-ordinates of target
+        azel=target.azel(timestamp=j, antenna=refant)
+        # apply refraction to target co-ordinates (already in radians)
+        ref_corr_el = rc.apply(azel[1], temp, pres, humi)
+        # Construct target object
+        trgt = katpoint.construct_azel_target(azel[0], ref_corr_el)
+        # Get offset az/el co-ordinates (direction in which reference antenna is pointing)
+        az = ts.get_range(str(refant.name)+'_pos_actual_scan_azim', et=j )
+        el = ts.get_range(str(refant.name)+'_pos_actual_scan_elev', et=j )
+        az_actual = katpoint.deg2rad(az[0][0])
+        el_actual = rc.apply(katpoint.deg2rad(el[0][0]), temp, pres, humi)
+        # Project spherical coordinates to plane with target position as reference
+        offset = trgt.sphere_to_plane(az_actual,el_actual, coord_system='azel', antenna=refant, timestamp=j)
+        offset = (katpoint.rad2deg(offset[0]),katpoint.rad2deg(offset[1]))
+        offsets.append(offset)
+
+    return offsets
+
 
 def flush_pipeline(ts, parameters, solution_stores):
-    """Flush pending calibration products in pipeline.
+    """Secondary pipeline calibration.
```

Comment:
This is not very descriptive. Let's try another version:
```suggestion
    """Complete any unfinished calibration products remaining in the pipeline.
```